src/AccessControl.js

```diff
@@ -0,0 +1,167 @@
+import { useQueryClient } from '@tanstack/react-query';
+import { abi, address } from '@vderunov/whitelist-contract/deployments/11155420/Whitelist';
+import { Contract, ethers } from 'ethers';
+import { useState } from 'react';
+import usePermissions from './usePermissions';
+import { useSynthetix } from './useSynthetix';
+
+function AccessControl() {
+  const queryClient = useQueryClient();
+  const [synthetix] = useSynthetix();
+  const permissions = usePermissions();
+  const [userApproveWallet, setUserApproveWallet] = useState('');
+  const [userApproveWalletError, setUserApproveWalletError] = useState('');
+  const [userRevokeWallet, setUserRevokeWallet] = useState('');
+  const [userRevokeWalletError, setUserRevokeWalletError] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [loadingText, setLoadingText] = useState('');
+  const contract = new Contract(address, abi, synthetix.signer);
+
+  const executeContractMethod = async (method, errorHandlerMsg) => {
```

Comment:
all of the async functions here can (and should) become useQuery or useMutation :)

Reply:
Added separate hooks for requests to the contract:

- `useApplyForWhitelistMutation`
- `useApproveApplicationMutation`
- `useRenounceAssignedRoleMutation`
- `useRevokeAccessMutation`